File: nao/doc-release-1.14-public/_downloads/alleds_fade.py
# -*- encoding: UTF-8 -*-

import sys
from naoqi import ALProxy

if (len(sys.argv) < 2):
    print "Usage: 'python leds_fade.py IP [PORT]'"
    sys.exit(1)

IP = sys.argv[1]
PORT = 9559
if (len(sys.argv) > 2):
    PORT = sys.argv[2]
try:
    proxy = ALProxy("ALLeds", IP, PORT)
except Exception,e:
    print "Could not create proxy to ALLeds"
    print "Error was: ",e
    sys.exit(1)

# Example showing how to fade the ears group to mid-intensity
name = 'EarLeds'
intensity = 0.5
duration = 1.0
proxy.fade(name, intensity, duration)

File: nao/doc-release-1.14-public/_downloads/alleds_fade.cpp
#include <iostream>
#include <alproxies/alledsproxy.h>

int main(int argc, char **argv)
{
  if (argc < 2) {
    std::cerr << "Usage: leds_ledsfade pIp" << std::endl;
    return 1;
  }
  const std::string pIp = argv[1];

  // Create a proxy to ALLeds.
  AL::ALLedsProxy leds(pIp);

  // Example showing how to fade the left eye group to mid-intensity
  std::string name = "EarLeds";
  float intensity = 0.5f;
  float duration = 1.0f;
  leds.fade(name, intensity, duration);
  std::cout << "Faded " << name << " to " << intensity << " in " << duration
            << std::endl;

  return 0;
}
